runtime/common/src/parachains.rs

```diff
@@ -910,7 +905,8 @@ impl<T: Trait> Module<T> {
 		let sorted_validators = make_sorted_duties(&duty_roster.validator_duty);
 
 		let parent_hash = <system::Module<T>>::parent_hash();
-		let localized_payload = |statement: Statement| localized_payload(statement, parent_hash);
+		let session_index = <session::Module<T>>::current_index();
+		let localized_payload = |statement: Statement| localized_payload(statement, session_index.encode(), parent_hash);
```

Comment:
It would be better to have all encoding happen within the `localized_payload` function.